test/test_composability.py

```diff
@@ -339,14 +349,23 @@ def build_stage(stage_idx, num_stages):
         # Validate that whichever weights we have locally match that part of our local/full ref model
         # (we force FSDP's grads to be all-gathered (.full_tensor) to make it simpler)
         ref_parameters = dict(ref_model.named_parameters())
-        for partial_model, offset in zip(partial_models, offsets):
-            for name, p in partial_model.named_parameters():
-                parts = name.split(".")
-                parts[0] = str(int(parts[0]) + offset)
-                name = ".".join(parts)
-                ref_p = ref_parameters[name]
-                self.assertTrue(isinstance(p.grad, DTensor))
-                self.assertEqual(ref_p.grad, p.grad.full_tensor())
+        if dp_type == "FSDP":
+            for partial_model, offset in zip(partial_models, offsets):
+                for name, p in partial_model.named_parameters():
+                    parts = name.split(".")
+                    parts[0] = str(int(parts[0]) + offset)
+                    name = ".".join(parts)
+                    ref_p = ref_parameters[name]
+                    self.assertTrue(isinstance(p.grad, DTensor))
+                    self.assertEqual(ref_p.grad, p.grad.full_tensor())
+        elif dp_type == "DDP":
+            for partial_model, offset in zip(partial_models, offsets):
+                for name, p in partial_model.named_parameters():
+                    parts = name.split(".")[1:]  # remove the "module." prefix
+                    parts[0] = str(int(parts[0]) + offset)
+                    name = ".".join(parts)
+                    ref_p = ref_parameters[name]
+                    self.assertEqual(ref_p.grad, p.grad)
```

Comment:
nit: not required for this PR, is there a better way to access params/sublayers now?
Cc: I thought @wconstab found something like `ModuleDict`.